File: Sample Code/ObjcSampleCode/DJISdkDemo/Demo/DemoUtility/UIUtility/DemoSettingItem.m
//
//  SettingItem.m
//  DJISdkDemo
//
//  Copyright © 2015 DJI. All rights reserved.
//

#import "DemoSettingItem.h"

@implementation DemoSettingItem

-(id) initWithName:(NSString *)name andClass:(Class)viewControllerClass {
    self = [super init];
    if (self) {
        _itemName = name;
        _viewControllerClass = viewControllerClass;
    }
    
    return self;
}

+(id)itemWithName:(NSString *)name andClass:(Class)viewControllerClass {
    return [[DemoSettingItem alloc] initWithName:name andClass:viewControllerClass];
}

@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/DemoUtility/UIUtility/DemoSettingItem.swift
//
//  DemoSettingItem.h
//  DJISdkDemo
//
//  Created by DJI on 12/17/15.
//  Copyright © 2015 DJI. All rights reserved.
//
import Foundation
import UIKit

struct DemoSettingItem {
    var itemName: String = ""
    var viewControllerClass:UIViewController.Type? = nil
    
    init(name: String, andClass viewControllerClass: UIViewController.Type?) {
        self.itemName = name
        self.viewControllerClass = viewControllerClass
    }
}